src/de.rs

```diff
@@ -36,7 +36,7 @@ impl<'a, T: DeserializeParams<'a>> Uri<'a, bitcoin::address::NetworkUnchecked, T
             None => (string, None),
         };
 
-        let address = address.parse().map_err(Error::uri)?;
+        let address = address.parse().map_err(|e| Error::Uri(UriError(UriErrorInner::Address(e))))?;
```

Comment:
not sure whats the root cause of this, but i suspect its something with the new error `ParseError` we are using that is not implementing everything the `Error` is

Reply:
What? I tried to use `Error::uri` and it works fine.

Reply:
Ha 
Which rust version? 
Can u suggest the code that worked for you?

Reply:
Both 1.48.1 and 1.63, the original line: `let address = address.parse().map_err(Error::uri)?;`

What exact error are you getting? BTW 0.31 raised MSRV to 1.48, so we need to bump that one in README as well.

Reply:
Hello, thanks for the review!

so for some reason it was not working and after testing it again with @DanGould it seems to be working just fine with the previous line. 
I fixed up a commit to revert this and also added a commit to bump msrv to `1.48`